icicle/include/api/gen_c_api.py

```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/env python3
+"""
+This script generates extern declarators for every curve/field
+"""
+
+from pathlib import Path
+
+API_PATH = Path(__file__).resolve().parent
+TEMPLATES_PATH = API_PATH.joinpath("templates")
+
+CURVES_MODULES = {
+    "curve",
+    "msm",
+    "ecntt"
+}
+
+FIELDS_MODULES = {
+    "field",
+    "ntt",
+    "poseidon",
+}
+
+CURVES_CONFIG = {
```

Comment:
please explain what the dicts for curves and fields are. I can see it from the code below but would be clearer to have it explained here